pkg/httpapi/api.go

```diff
@@ -161,6 +166,29 @@ func (a *APIRouter) GetApplication(w http.ResponseWriter, r *http.Request) {
 	marshalResponse(w, appEnvironments)
 }
 
+func (a *APIRouter) ListApplications(w http.ResponseWriter, r *http.Request) {
+	appList := &argoV1aplha1.ApplicationList{}
+	var listOptions []ctrlclient.ListOption
+
+	listOptions = append(listOptions, ctrlclient.InNamespace(""))
+	listOptions = append(listOptions, ctrlclient.MatchingLabels{"app.kubernetes.io/managed-by": "kam"})
```

Comment:
We need a `label` to identify the applications deployed using KAM. we want to ignore applications deployed on the cluster without the label at least for now. 
So I have added a provisional label `app.kubernetes.io/managed-by: kam`, but I am not sure if it's best suited!! 
Any thoughts on this ?

Reply:
I think we might need to still accept the gitops repo URL as a parameter.  eg. Return all the apps from that configuration back to the UI.   Eg. /applications?url=http://mygitopsrepo

This is similar to the pipelines api.